cranelift-codegen/meta/src/cdsl/ast.rs

```diff
@@ -92,101 +91,82 @@ impl DefPool {
 }
 
 #[derive(Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct DefIndex(u32);
+pub(crate) struct DefIndex(u32);
 entity_impl!(DefIndex);
 
-#[derive(Debug, Clone)]
-enum LiteralValue {
+#[derive(Clone, Debug)]
+pub(crate) struct LiteralEnum {
+    rust_type: String,
+    value: &'static str,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct LiteralBits {
+    rust_type: String,
+    value: u64,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) enum Literal {
     /// A value of an enumerated immediate operand.
     ///
     /// Some immediate operand kinds like `intcc` and `floatcc` have an enumerated range of values
     /// corresponding to a Rust enum type. An `Enumerator` object is an AST leaf node representing one
     /// of the values.
-    Enumerator(&'static str),
+    Enumerator(LiteralEnum),
```

Comment:
Readability, usability and maintenance. It's harder to mix the meaning of two strings when you have named fields. For a struct with one field, I'm usually happy to use an anonymous struct, since no confusion is possible. If you feel strongly about it, I can just use anonymous struct, though.

Reply:
For clarification, anonymous struct can have name fields:

```rust
enum Literal {
  Enumerator{ rust_type: String, value: &'static str },
}
```

Reply:
TIL, thanks.